File: go/sum-of-multiples/sum_of_multiples.go
// Package summultiples sums multiples.
package summultiples

func isDiv(n int, d []int) bool {
	for _, i := range d {
		if i == 0 {
			continue
		}
		if n%i == 0 {
			return true
		}
	}
	return false
}

// SumMultiples sums up multiples of the divisors.
func SumMultiples(lim int, div ...int) int {
	var sum int
	for i := 1; i < lim; i++ {
		if isDiv(i, div) {
			sum += i
		}
	}
	return sum
}

File: python/sum-of-multiples/sum_of_multiples.py
def sum_of_multiples(limit, multiples):
  multiples = [m for m in multiples if m > 0]
  return sum(
    i for i in range(1, limit)
    if any(i % f == 0 for f in multiples))


# vim:ts=2:sw=2:expandtab
